OPHD/UI/Core/RadioButtonGroup.cpp

```diff
@@ -0,0 +1,78 @@
+#include "RadioButtonGroup.h"
+
+#include "../../Cache.h"
+#include "../../Constants.h"
+
+#include <NAS2D/Utility.h>
+#include <NAS2D/Renderer/Renderer.h>
+#include <NAS2D/MathUtils.h>
+
+#include <algorithm>
+
+
+using namespace NAS2D;
+
+
+RadioButtonGroup::RadioButtonGroup(std::vector<ButtonInfo> buttonInfos)
+{
+	mRadioButtons.reserve(buttonInfos.size());
+	for(auto &buttonInfo : buttonInfos)
+	{
+		NAS2D::Vector<int> offset = {0, 13};
+		offset.y = mRadioButtons.size() * offset.y;
+
+		auto &button = mRadioButtons.emplace_back(buttonInfo.name, this, buttonInfo.delegate);
+		button.visible(visible());
+		button.position(mRect.startPoint() + offset);
+	}
+}
+
+void RadioButtonGroup::onMove(NAS2D::Vector<int> displacement)
+{
+	Control::onMove(displacement);
+
+	for (auto &control : mRadioButtons)
+	{
+		control.position(control.position() + displacement);
+	}
+}
+
+void RadioButtonGroup::update()
+{
+	if (!visible()) { return; }
+
+	if( mRadioButtons.size() > 0 && !hasOneSelectedItem() ) {
+		clearSelection();
+		mRadioButtons[0].click();
+	}
```

Comment:
I feel like this part could potentially be improved.

I feel like the `RadioButtonGroup` class should maintain an invariant of having at most one button selected. (No buttons selected should be fine too). The exposed methods should make sure this is always true, and so there would be no need to check this in the `update()` method.

Reply:
Changed the implementation accordingly.